apps/studio/components/interfaces/Auth/AuthProvidersForm/FormField.tsx

```diff
@@ -1,288 +1,276 @@
-  const [hidden, setHidden] = useState(!!properties.isSecret)
-  const [dateAsText, setDateAsText] = useState(
-    formValues[name] ? formatDate(new Date(formValues[name])) : ''
-  )
-
-  useEffect(() => {
-    if (properties.show && properties.show.key && !formValues[properties.show.key]) {
-      setFieldValue(name, '')
-      setDateAsText('')
-    }
-  }, [properties.show && properties.show.key && !formValues[properties.show.key]])
-
-  if (properties.show) {
-    if (properties.show.matches) {
-      if (!properties.show.matches.includes(formValues[properties.show.key])) {
-        return null
-      }
-    } else if (!formValues[properties.show.key]) {
-      return null
-    }
-  }
+export const FormField = ({ name, properties, control, disabled = false }: FormFieldProps) => {
+  const [hidden, setHidden] = useState(properties.type === `string` && !!properties?.isSecret)
 
-  switch (properties.type) {
-    case 'datetime':
-      return (
-        <Input
-          size="small"
-          layout="vertical"
-          id={name}
-          name={name}
-          type="text"
-          value={dateAsText}
-          readOnly
-          label={properties.title}
-          labelOptional={
-            properties.descriptionOptional ? (
-              <ReactMarkdown unwrapDisallowed disallowedElements={['p']}>
-                {properties.descriptionOptional}
-              </ReactMarkdown>
-            ) : null
-          }
-          descriptionText={
-            properties.description ? (
-              <ReactMarkdown unwrapDisallowed disallowedElements={['p']}>
-                {properties.description}
-              </ReactMarkdown>
-            ) : null
-          }
-          actions={
-            <DatePicker
-              selectsRange={false}
-              minDate={new Date()}
-              from={formValues[name]}
-              to={formValues[name]}
-              onChange={(date) => {
-                if (date && date.to) {
-                  setFieldValue(name, date.to)
-                  setDateAsText(formatDate(new Date(date.to)))
-                } else {
-                  setDateAsText('')
-                  setFieldValue(name, '')
+  return (
+    <FormField_Shadcn_
+      key={name}
+      name={name}
+      control={control}
+      render={({ field }) => {
+        switch (properties.type) {
+          case 'datetime':
+            return (
+              <FormItemLayout
+                name={name}
+                layout="vertical"
+                label={properties.title}
+                labelOptional={
+                  properties.descriptionOptional ? (
+                    <Markdown
+                      content={properties.descriptionOptional}
+                      className="text-foreground-lighter"
+                    />
+                  ) : null
                 }
-              }}
-            >
-              <span>Pick</span>
-            </DatePicker>
-          }
-        />
```

Comment:
this is the only datepicker + text input combo we have like this in the entire studio app. It seemed easier to me for wiring up this to the form to use the datepicker popover button text to display the current value rather than this dummy input

open to suggestions if someone has a better idea of how to do this, but for the moment I just want to get this in a testable state